app/celery/process_ses_receipts_tasks.py

```diff
@@ -1,80 +1,100 @@
+        # TODO remove after smoke testing on prod is implemented
+        current_app.logger.info(f"Attempting to process SES delivery status message from SNS with type: {notification_type} and body: {ses_message}")
         bounce_message = None
 
         if notification_type == 'Bounce':
-            notification_type, bounce_message = determine_notification_bounce_type(notification_type, ses_message)
+            bounce_message = determine_notification_bounce_type(ses_message)
         elif notification_type == 'Complaint':
             _check_and_queue_complaint_callback_task(*handle_complaint(ses_message))
             return True
 
-        aws_response_dict = get_aws_responses(notification_type)
+        aws_response_dict = get_aws_responses(ses_message)
 
-        notification_status = aws_response_dict['notification_status']
-        reference = ses_message['mail']['messageId']
+        notification_status = aws_response_dict["notification_status"]
+        reference = ses_message["mail"]["messageId"]
 
         try:
-            notification = notifications_dao.dao_get_notification_or_history_by_reference(reference=reference)
+            notification = notifications_dao.dao_get_notification_by_reference(reference)
         except NoResultFound:
-            message_time = iso8601.parse_date(ses_message['mail']['timestamp']).replace(tzinfo=None)
+            message_time = iso8601.parse_date(ses_message["mail"]["timestamp"]).replace(tzinfo=None)
             if datetime.utcnow() - message_time < timedelta(minutes=5):
                 current_app.logger.info(
-                    f"notification not found for reference: {reference} (update to {notification_status}). "
+                    f"notification not found for reference: {reference} (while attempting update to {notification_status}). "
                     f"Callback may have arrived before notification was persisted to the DB. Adding task to retry queue"
                 )
                 self.retry(queue=QueueNames.RETRY)
             else:
                 current_app.logger.warning(
-                    f"notification not found for reference: {reference} (update to {notification_status})"
+                    "notification not found for reference: {} (while attempting update to {})".format(reference, notification_status)
                 )
             return
 
         if bounce_message:
             current_app.logger.info(f"SES bounce for notification ID {notification.id}: {bounce_message}")
 
-        if notification.status not in [NOTIFICATION_SENDING, NOTIFICATION_PENDING]:
+        if notification.status not in {NOTIFICATION_SENDING, NOTIFICATION_PENDING}:
             notifications_dao._duplicate_update_warning(
-                notification=notification,
-                status=notification_status
+                notification, 
+                notification_status
             )
             return
+
+        notifications_dao._update_notification_status(
```

Comment:
[python newbie question] does it mean anything when a method starts with a `_`? Either enforced by the language or by convention.

Reply:
No enforcement that I know of, but by convention these are often used to differentiate public from private/internal methods at a glance. Looks like this is an example where it was originally written as an internal method, but then started to get accessed elsewhere and was never renamed

Reply:
:eye-twitching: